# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/scene_building.rs
# Commit: 6fc108de5c8f
# Full Hash: 6fc108de5c8fd527d0f5b6ae7b91604cc22e7d62
# Author: Nicolas Silva <nsilva@mozilla.com>
# Date: 2021-05-03 15:32:34
# Description:
#   Bug 1707744 - Avoid far gradient endpoints causing large gradient segments. r=gfx-reviewers,lsalzman
#   
#   Large segment bounds trip an assertion when casting coordinates to integers. Clipping early also reduces the amount of cached pixels.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113819
# ==============================================================================

diff -r 5d436e803f68 -r 6fc108de5c8f gfx/wr/webrender/src/scene_building.rs
--- a/gfx/wr/webrender/src/scene_building.rs	Mon May 03 09:12:42 2021 +0000
+++ b/gfx/wr/webrender/src/scene_building.rs	Mon May 03 09:12:42 2021 +0000
@@ -1244,8 +1244,8 @@
                     &mut end,
                     info.gradient.extend_mode,
                     &mut stops,
-                    &mut |rect, clip, start, end, stops| {
-                        let layout = LayoutPrimitiveInfo { rect: *rect, clip_rect: *clip, flags };
+                    &mut |rect, start, end, stops| {
+                        let layout = LayoutPrimitiveInfo { rect: *rect, clip_rect: *rect, flags };
                         if let Some(prim_key_kind) = self.create_linear_gradient_prim(
                             &layout,
                             start,